### PR TITLE
Created a more charming example out of the supplementary views exampl…

### PR DESCRIPTION
**Pull request by weyert**
…e so it uses sticky headers

## Changes in this pull request

Issue fixed: #831 

### Checklist

- [X] All tests pass. Demo project builds and runs.
- [ ] I added tests, an experiment, or detailed why my change isn't tested.
- [ ] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [X] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: Examples/Examples-iOS/IGListKitExamples/ViewControllers/SupplementaryViewController.swift [hunk 1]
```diff
@@ -20,17 +20,50 @@ final class SupplementaryViewController: UIViewController, ListAdapterDataSource
...
+            "What's the eta?"
+            ])
     ]
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        if let flowLayout = collectionView.collectionViewLayout as? ListCollectionViewLayout {
+            let statusBarSize = UIApplication.shared.statusBarFrame.size
+            let navigationBarVisible = self.navigationController?.isNavigationBarHidden == false
+            let navigationBarHeight = ( navigationBarVisible ? self.navigationController!.navigationBar.frame.height : 0 )
```
Comment (rnystrom) [inline Examples/Examples-iOS/IGListKitExamples/ViewControllers/SupplementaryViewController.swift]: Any chance we can remove reaching into the nav and status bar frames? We don't want to recommend doing this as its not really future-friendly. What about the top layout guide?

Comment (weyert) [inline Examples/Examples-iOS/IGListKitExamples/ViewControllers/SupplementaryViewController.swift]: @rnystrom Sure, I will see how I can change the code to make it work without accessing the nav/status

**Comment by weyert** [comment]
@rnystrom Thank you for suggesting topLayoutGuide. Sadly, it's zero in `viewDidLoad` because the view hasn't been drawn yet. I have moved it to `viewDidAppear`

**Comment by rnystrom** [comment]
@weyert another idea (and I honestly have no idea how good a practice this is...) is to hand the _layout guide itself_ to the section controller. Then, come sizing time, the SC can ask for the `length`.

I came up w/ this idea on my GitHub app. [Here](https://github.com/rnystrom/Freetime/blob/master/Classes/Notifications/NotificationsViewController.swift#L175) is where I pass in the layout guide, and [here](https://github.com/rnystrom/Freetime/blob/master/Classes/Notifications/NoNewNotificationsSectionController.swift#L26) is where I use its value.

The end result is a cell that fits perfectly within the remaining height.

What do you think?

**Comment by weyert** [comment]
@rnystrom I would be happy to implement it. Only wouldn't it make the code of this specific example a bit more complicated then really necessary? It's your call, let me know :)

**Comment by weyert** [comment]
@rnystrom I have had a look at your suggestion again but I don't think it would useful here. For the sticky headers we need to set a offset in `IGListKit`'s layout so that sticks below the navigation bar instead of under it. Looking at your code in `FreeTime` we won't be able to use that approach while it's great idea! The problem is that it's a collection view layout property and you can not access that through `collectionContext` but I also think it's not a great idea to keep changing the layout properties between section controllers.

**Comment by weyert** [comment]
How shall we progress this?

**Comment by rnystrom** [comment]
@weyert I'm going to close this out, I want to make sure our examples are really lightweight, and having to explain what is happening beyond code might be a little confusing. I think what we have now are pretty good examples using our custom layout. Lmk if you disagree and think this is going to be really useful.

**Comment by weyert** [comment]
Sounds good to me! Just wanted let one example use the sticky headers :-)

**Pull request closed without merging**